### PR TITLE
feat: Update template to include `DashboardPlugin`

### PR DESCRIPTION
**Pull request by mofojed**
- Template now includes `DashboardPlugin` and registering a custom panel
- Uses the new `useComponent` hook provided from DHC
- Needs https://github.com/deephaven/web-client-ui/pull/1409 to merge first

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,48 +14,68 @@
...
-    "image-webpack-loader": "^8.0.1",
+    "eslint": "^8.29.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-prettier": "^5.0.0",
+    "eslint-plugin-react": "^7.28.0",
     "jest": "^27.4.5",
-    "node-sass": "^7.0.1",
-    "prettier": "^2.2.1",
+    "prettier": "^3.0.0",
+    "process": "^0.11.10",
```
Comment (mattrunyon) [inline package.json]: Not sure why this is needed as a dependency

File: package.json [hunk 2]
```diff
@@ -14,48 +14,68 @@
...
     "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-hot-loader": "^4.13.0",
```
Comment (mattrunyon) [inline package.json]: Don't think this is needed either

File: package.json [hunk 3]
```diff
@@ -14,48 +14,68 @@
...
+    "@deephaven/components": "^0.46.0",
+    "@deephaven/dashboard": "^0.46.0",
+    "@deephaven/dashboard-core-plugins": "^0.46.0",
+    "@deephaven/golden-layout": "^0.46.0",
+    "@deephaven/grid": "^0.46.0",
+    "@deephaven/iris-grid": "^0.46.0",
+    "@deephaven/jsapi-types": "^0.46.0",
+    "@deephaven/jsapi-utils": "^0.46.0",
+    "@deephaven/log": "^0.46.0",
+    "@deephaven/redux": "^0.46.0",
```
Comment (mattrunyon) [inline package.json:74]: These all aren't necessarily peer deps, but I don't think it matters for this template right now

File: src/CustomPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { DashboardPanelProps } from '@deephaven/dashboard';
+import Log from '@deephaven/log';
+import './CustomPanel.scss';
```
Comment (mattrunyon) [inline src/CustomPanel.tsx]: Does this actually work? I had issues w/ it in another plugin and had to use [this](https://github.com/deephaven/deephaven-js-plugins/blob/main/plugins/dashboard-object-viewer/src/DashboardPlugin/DashboardPlugin.tsx#L9) instead

**Review by mattrunyon — CHANGES_REQUESTED** [review]

File: src/CustomPanel.tsx (hunk 4 above)
Comment (mofojed) [reply src/CustomPanel.tsx]: I think it worked? Pretty sure I tested it with some odd styling...

File: src/DashboardPlugin.tsx [hunk 5]
```diff
@@ -1,11 +1,66 @@
...
+import React, { useCallback } from 'react';
+import {
+  DashboardPluginComponentProps,
+  useDashboardPanel,
+  useListener,
+} from '@deephaven/dashboard';
+import Log from '@deephaven/log';
+import CustomPanel from './CustomPanel';
+import { IrisGridEvent } from '@deephaven/dashboard-core-plugins';
+import { IrisGridPanel } from '@deephaven/dashboard-core-plugins/dist/panels/IrisGridPanel';
```
Comment (mofojed) [inline src/DashboardPlugin.tsx]: Importing from /dist/ :(

File: src/CustomPanel.tsx (hunk 4 above)
Comment (mofojed) [reply src/CustomPanel.tsx]: It indeed does not work. Trying with `?inline`

Comment (mofojed) [reply src/CustomPanel.tsx]: Switched to `?inline`

**Review by mattrunyon — APPROVED** [review]

**Pull request merged**